### PR TITLE
refactor(DataList): Refactor DataList based on changes in core

### PR DESCRIPTION
**Pull request by tlabaj**
#1747

**What**:

**Additional issues**:

### CODE REVIEW

File: packages/patternfly-4/react-core/src/components/DataList/DataListAction.d.ts [hunk 1]
```diff
@@ -1,10 +1,11 @@
 import { FunctionComponent, HTMLProps, ReactNode } from 'react';
 
 export interface DataListActionProps extends HTMLProps<HTMLDivElement> {
+  children: any;
```
Comment (dlabaj) [inline packages/patternfly-4/react-core/src/components/DataList/DataListAction.d.ts]: This is required now.

File: packages/patternfly-4/react-core/src/components/DataList/DataListAction.d.ts [hunk 2]
```diff
@@ -1,10 +1,11 @@
...
   'aria-labelledby': string;
   'aria-label': string;
   id: string;
-  actions: ReactNode[];
+  actions?: ReactNode[];
```
Comment (dlabaj) [inline packages/patternfly-4/react-core/src/components/DataList/DataListAction.d.ts]: This should be removed.

File: packages/patternfly-4/react-core/src/components/DataList/DataListCheck.js [hunk 3]
```diff
@@ -4,15 +4,17 @@ import PropTypes from 'prop-types';
...
-  <div className={css('pf-c-data-list__check', className)}>
-    <input
-      {...props}
-      type="checkbox"
-      onChange={event => onChange(event.currentTarget.checked, event)}
-      aria-invalid={!isValid}
-      disabled={isDisabled}
-      checked={isChecked || checked}
-    />
+  <div className={css(styles.dataListItemControl, className)} {...props}>
```
Comment (jschuler) [inline packages/patternfly-4/react-core/src/components/DataList/DataListCheck.js]: props are spread twice here and in input

File: packages/patternfly-4/react-core/src/components/DataList/index.d.ts [hunk 4]
```diff
@@ -1,7 +1,8 @@
 export { default as DataList } from './DataList';
-export { default as DataListItem } from './DataListItem';
+export { default as DataListAction } from './DataListAction';
```
Comment (jschuler) [inline packages/patternfly-4/react-core/src/components/DataList/index.d.ts]: We usually also export out the *Props, e.g. DataListActionProps

File: packages/patternfly-4/react-core/src/components/DataList/DataListItemCells.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+  <div className={css(styles.dataListItemContent, className)} {...props}>
+    {dataListCells}
+  </div>
+);
+
+DataListItemCells.propTypes = {
+  /** Additional classes added to the DataList item Content Wrapper.  Children should be one ore more <DataListCell> nodes */
+  className: PropTypes.string,
+  /** Array of <DataListCell> nodes that are rendered one after the other. */
+  dataListCells: PropTypes.arrayOf(PropTypes.node).isRequired,
```
Comment (jschuler) [inline packages/patternfly-4/react-core/src/components/DataList/DataListItemCells.js]: We already compose
```
<DataList>
  <DataListItem>
      <DataListItemCells />
  </DataListItem>
</DataList>
```
Then why not also `<DataListCell>` into DataListItemCells?

**Comment by christiemolloy** [comment]
Also looks like you included the third data list item within the second data list, thats why they're appearing side by side like this:

Screen Shot 2019-04-12 at 9 36 27 AM

**Comment by christiemolloy** [comment]
You'll also need to fix the other examples, it goes:

`pf-c-data-list` 
       `pf-c-data-list__item` 
              `pf-c-data-list__item-row`
                    `pf-c-data-list_item-control`
                    `pf-c-datalist__item-content`
                          `pf-c-data-list__cell`
                     `pf-c-data-list__item-action`
            `pf-c-data-list__item-expandable-content`
                      `pf-c-data-list__item-expandable-content-body`

File: packages/patternfly-4/react-core/src/components/DataList/index.d.ts (hunk 4 above)
Comment (tlabaj) [reply packages/patternfly-4/react-core/src/components/DataList/index.d.ts]: So add it for all of t hem correct?  even the ones that were not updated?

File: packages/patternfly-4/react-core/src/components/DataList/DataListItemCells.js (hunk 5 above)
Comment (tlabaj) [reply packages/patternfly-4/react-core/src/components/DataList/DataListItemCells.js]: I want to enforce the consumer using DataListItemsCell wrapper around the DataListItems.  That is why I have the prop.  Does that answer you question?  I am not sure I understand it.

Comment (jschuler) [reply packages/patternfly-4/react-core/src/components/DataList/DataListItemCells.js]: Consumer could still pass anything they want into the dataListCells prop... why not
```
<DataListItemCells>
  <DataListCell>...</DataListCell>
</DataListItemCells>
```

**Pull request merged**